File: integrations/aspnetcore/src/Scalar.AspNetCore/Enums/TagSorter.cs
using System.ComponentModel;
using System.Text.Json.Serialization;
using NetEscapades.EnumGenerators;

namespace Scalar.AspNetCore;

/// <summary>
/// Specifies the sorting options for tags in the Scalar API reference.
/// </summary>
[EnumExtensions]
[JsonConverter(typeof(TagSorterJsonConverter))]
public enum TagSorter
{
    /// <summary>
    /// Sort tags alphabetically.
    /// </summary>
    [Description("alpha")]
    Alpha
}

File: integrations/java/webjar/src/main/java/com/scalar/maven/webjar/enums/TagSorter.java
package com.scalar.maven.webjar.enums;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

/**
 * Represents the sorting method for tags in the Scalar API Reference.
 */
public enum TagSorter {
    /**
     * Sort tags alphabetically.
     */
    ALPHA("alpha");

    private final String value;

    TagSorter(String value) {
        this.value = value;
    }

    /**
     * Creates a TagSorter from a string value.
     *
     * @param value the string value
     * @return the corresponding TagSorter
     * @throws IllegalArgumentException if the value is not recognized
     */
    @JsonCreator
    public static TagSorter fromValue(String value) {
        for (TagSorter sorter : values()) {
            if (sorter.value.equals(value)) {
                return sorter;
            }
        }
        throw new IllegalArgumentException("Unknown tag sorter: " + value);
    }

    /**
     * Gets the string value for JSON serialization.
     *
     * @return the string value
     */
    @JsonValue
    public String getValue() {
        return value;
    }
}
